runtime/worker.rs

```diff
@@ -57,6 +57,17 @@ impl ExitCode {
     self.0.store(code, Relaxed);
   }
 }
+
+/// Clear extension modules from the module map, except preserve `ext:deno_node`
+/// modules as `node:` specifiers.
+pub fn init_runtime_module_map(js_runtime: &mut JsRuntime) {
+  js_runtime.clear_module_map(
+    deno_node::SUPPORTED_BUILTIN_NODE_MODULES
```

Comment:
I think this rename and retain operation callback should live in the extension definition macro (deno_core::extension!) itself.

Reply:
> I think this rename and retain operation callback should live in the extension definition macro (deno_core::extension!) itself.

That was my first idea. The problem is we don't want this step to be taken when we're creating the intermediary `RUNTIME_SNAPSHOT` since the CLI extensions will want internal access. While thinking of how on earth I'd document or option-ise this very specific behaviour, it became clear to me that it's not a property of the extension but rather a bespoke operation called by the host only at the final step. So I added a procedural binding for it.

IMO It turns out that this solution is the less coupled one and is easier to delete if e.g. we figure out a better cross-crate JS extension interface.

Reply:
That makes sense to me. I think we can take another look at the runtime/realm setup process after this lands (especially if we can get modules working correctly in realms soon).